compiler-core/src/type_/error.rs

```diff
@@ -322,6 +322,10 @@ pub enum Warning {
         location: SrcSpan,
         name: SmolStr,
     },
+
+    DoubleUnary {
```

Comment:
Can we call this something like PointlessDoubleNegation? Make it clear what the error is. Other unary operators could be valid